File: bounce-game.java
import java.util.concurrent.ThreadLocalRandom;

class Main {
  public static void main(String[] args) {
    int pos = ThreadLocalRandom.current().nextInt(1, 10 + 1);
    Boolean backwards = false;

    for(int i = 1; i <= 100; i++){
      String field = "!          !";
      field = field.substring(0,pos) + 'O' + field.substring(pos+1);
      
      if(pos == field.length() - 2) {
        backwards = true;
      } else if(pos == 1) {
        backwards = false;
      }
      System.out.println(field);

      if(backwards){
        pos--;
      } else {
        pos++;
      }
    }
  }
}

File: bounce-game.go
package main

import (
	"fmt"
  "math/rand"
  "time"
)

func main() {
  s1 := rand.NewSource(time.Now().UnixNano())
  r1 := rand.New(s1)

  var pos = r1.Intn(3) + 1;
  var backwards = false;

  for i := 1; i < 90; i++ {
    var field = "!          !";

    field = field[:pos] + "O" + field[pos+1:];

    if pos == len(field) - 2 {
      backwards = true;
    } else if pos == 1 {
      backwards = false;
    }
    fmt.Println(field);

    if backwards {
      pos--;
    } else {
      pos++;
    }
  }
}